File: Python/Vehicle.py
class Vehicle:
    def __init__(self, plate_number, brand, year, color):
        # Konstruktor kelas Vehicle, menerima parameter plat nomor, merk, tahun produksi, dan warna
        self.__plate_number = plate_number   # Inisialisasi plat nomor
        self.__brand = brand                 # Inisialisasi merk
        self.__year = year                   # Inisialisasi tahun produksi
        self.__color = color                 # Inisialisasi warna

    def get_plate_number(self):
        # Metode untuk mendapatkan plat nomor kendaraan
        return self.__plate_number

    def set_plate_number(self, plate_number):
        # Metode untuk mengatur plat nomor kendaraan
        self.__plate_number = plate_number

    def get_brand(self):
        # Metode untuk mendapatkan merk kendaraan
        return self.__brand

    def set_brand(self, brand):
        # Metode untuk mengatur merk kendaraan
        self.__brand = brand

    def get_year(self):
        # Metode untuk mendapatkan tahun produksi kendaraan
        return self.__year

    def set_year(self, year):
        # Metode untuk mengatur tahun produksi kendaraan
        self.__year = year

    def get_color(self):
        # Metode untuk mendapatkan warna kendaraan
        return self.__color

    def set_color(self, color):
        # Metode untuk mengatur warna kendaraan
        self.__color = color

File: cpp2/Vehicle.cpp
#include <iostream>
#include <string>

#ifndef Vehicle_H
#define Vehicle_H

class Vehicle {
private:
    std::string plate_number;
    std::string brand;
    int year;
    std::string color;

public:
    // Konstruktor untuk kelas Vehicle
    Vehicle(std::string plate_number, std::string brand, int year, std::string color)
        : plate_number(plate_number), brand(brand), year(year), color(color) {}

    // Getter dan setter untuk plat nomor
    std::string get_plate_number() const{
        return plate_number;
    }

    void set_plate_number(std::string plate_number) {
        this->plate_number = plate_number;
    }

    // Getter dan setter untuk merk
    std::string get_brand() const{
        return brand;
    }

    void set_brand(std::string brand) {
        this->brand = brand;
    }

    // Getter dan setter untuk tahun produksi
    int get_year() const{
        return year;
    }

    void set_year(int year) {
        this->year = year;
    }

    // Getter dan setter untuk warna
    std::string get_color() const{
        return color;
    }

    void set_color(std::string color) {
        this->color = color;
    }

};
#endif // VEHICLE_H